packages/react-native-gesture-handler/src/web/handlers/GestureHandler.ts

```diff
@@ -425,17 +442,27 @@ export default abstract class GestureHandler implements IGestureHandler {
 
     if (this.lastSentState !== newState) {
       this.lastSentState = newState;
-      invokeNullableMethod(onGestureHandlerStateChange, resultEvent);
+      invokeNullableMethod(
+        this.forReanimated
+          ? onGestureHandlerReanimatedStateChange!
+          : onGestureHandlerStateChange,
+        resultEvent
+      );
     }
     if (this.state === State.ACTIVE) {
       if (this.actionType !== ActionType.NATIVE_DETECTOR) {
         (resultEvent.nativeEvent as GestureHandlerNativeEvent).oldState =
           undefined;
       }
-      if (onGestureHandlerAnimatedEvent && this.dispatchesAnimatedEvents) {
-        invokeNullableMethod(onGestureHandlerAnimatedEvent, resultEvent);
-      }
-      invokeNullableMethod(onGestureHandlerEvent, resultEvent);
+
+      invokeNullableMethod(
+        this.forReanimated
+          ? onGestureHandlerReanimatedEvent!
+          : this.forAnimated
+            ? onGestureHandlerAnimatedEvent!
+            : onGestureHandlerEvent,
+        resultEvent
+      );
```

Comment:
Cosmetic thing, we could extract this to a separate variable `const method = ...` and then use `invokeNullableMethod(method, resultEvent)`. For me it looks cleaner, but I'll leave that up to you.